Detectors/PHOS/simulation/include/PHOSSimulation/RawWriter.h

```diff
@@ -94,21 +75,20 @@ class RawWriter
                       std::vector<char>& trailer, std::vector<char>& header) const;
 
  protected:
-  void createRawBunches(const std::vector<o2::phos::Digit*>& digits, std::vector<o2::phos::AltroBunch>& bunchHG,
+  void createRawBunches(short absId, const std::vector<o2::phos::Digit*>& digits, std::vector<o2::phos::AltroBunch>& bunchHG,
                         std::vector<o2::phos::AltroBunch>& bunchLG, bool& isLGFilled);
 
-  std::vector<int> encodeBunchData(const std::vector<int>& data);
+  std::vector<uint32_t> encodeBunchData(const std::vector<uint32_t>& data);
   void fillGamma2(float amp, float time, short* samples);
   // std::tuple<int, int, int> getOnlineID(int towerID);
   // std::tuple<int, int> getLinkAssignment(int ddlID);
 
-  ChannelHeader createChannelHeader(int hardwareAddress, int payloadSize);
   std::vector<char> createRCUTrailer(int payloadsize, int feca, int fecb, double timesample, double l1phase);
 
  private:
   FileFor_t mFileFor = FileFor_t::kFullDet;           ///< Granularity of the output files
   std::string mOutputLocation = "./";                 ///< Rawfile name
-  std::unique_ptr<o2::phos::Mapping> mMapping;        ///< Mapping handler
+  o2::phos::Mapping* mMapping = nullptr;              ///< Mapping handler
```

Comment:
what was the reason to change ``unique_ptr`` to raw pointer and use naked ``new`` in the its initialization below? The coding guidelines actually discourage nakes ``new``s.

Reply:
This pointer is not passed anywhere and used only within this class. I think there is no need to use more complicated object.

Reply:
The ``unique_ptr`` is not more complex than the raw pointer (in opposite to ``shared_ptr``) but it takes care of deletion: right now, since you use default destructor, the mMapping is not deleted (memory leak). With ``unique_ptr`` you are guaranteed to have if deleted by the default destructor.
The standard way (avoiding any new/delete) is:
````
std::unique_ptr<o2::phos::Mapping> mMapping;
...
...
mMapping = std::make_unique<o2::phos::Mapping>();

Reply:
Changed to unique_ptr where possible